lua/telescope/builtin/git.lua

```diff
@@ -351,8 +364,10 @@ local set_opts_cwd = function(opts)
   local use_git_root = utils.get_default(opts.use_git_root, true)
 
   if ret ~= 0 then
-    local output = utils.get_os_command_output({ "git", "rev-parse", "--is-inside-work-tree" }, opts.cwd)
-    if output[1] ~= "true" then
+    local in_worktree = utils.get_os_command_output({ "git", "rev-parse", "--is-inside-work-tree" }, opts.cwd)
+    local in_bare = utils.get_os_command_output({ "git", "rev-parse", "--is-bare-repository" }, opts.cwd)
+
+    if in_worktree[1] ~= "true" and in_bare[1] ~= "true" then
```

Comment:
I have an idea. We do this check once (the two commands) and of we are in a bare and not in a worktree (like if we cant execute it `find_files`, `status`) we set a flag in `opts` (`opts.is_bare`).

The builtins we want to disable (files, status) then have a function call at the top of the function (`validate_worktree(opts)`) that errors if its forbidden.

Does this make sense for you? That way we dont call `git` that often

Reply:
I think this is a much cleaner approach. I've refactored it to only call the git operations once, and then do a simple check for the `find_files` and `status`. Take a look and let me know what you think!

Reply:
Why do we need the `set_opts_bare` function. We could just do this here. after the if in 365. Otherwise we execute `git rev-pare --is-inside-work-tree` and `git rev-parse --is-bare-repository` twice again

Reply:
Yep sorry, didn't paying attention to the code I already wrote :laughing: - fixed now!